### PR TITLE
Convert climate water heaters to new water_heaters component

### PR DESCRIPTION
**Pull request by w1ll1am23**
This is to support this effort https://github.com/home-assistant/home-assistant/issues/13539 I have started working on this and have some of the initial work done. I am not a web dev so I don't really know what all is going on here, I just copied the climate stuff to water_heater. 

~~Currently this is kinda working, but not perfectly. I am getting the following error when I click to bring up the more-info~~

image

~~Please let me know if there is anything I am doing wrong or if you know how to fix the error.~~

~~*NOTE: I am not seeing the operation mode selection option which I am assuming is from the error? But I am not sure.~~

### CODE REVIEW

File: src/dialogs/more-info/controls/more-info-water_heater.js [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+      .single-row {
+        padding: 8px 0;
+      }
+      }
+    </style>
+
+    <div class$="[[computeClassNames(stateObj)]]">
+
+      <template is="dom-if" if="[[supportsOn(stateObj)]]">
```
Comment (balloob) [inline src/dialogs/more-info/controls/more-info-water_heater.js]: You call `supportsOn` here but you don't have that method defined.

**Comment by balloob** [comment]
The entity registry one is fine, it just means that the backend entity does not define a unique ID.

File: src/dialogs/more-info/controls/more-info-water_heater.js (hunk 1 above)
Comment (w1ll1am23) [reply src/dialogs/more-info/controls/more-info-water_heater.js]: Thanks, fixed.

**Comment by w1ll1am23** [comment]
@balloob I believe this is good to go.

**Review by balloob — APPROVED** [review]

**Comment by balloob** [comment]
Run `./node_modules/.bin/eslint src --fix` and commit the changes to fix lint changes 👍

**Comment by balloob** [comment]
This is ok to merge when water heaters have been added to the backend.

**Comment by w1ll1am23** [comment]
backend MR https://github.com/home-assistant/home-assistant/pull/17058

**Comment by balloob** [comment]
Going to merge this as it's not going to be in the way and the backend PR has emerged.

**Pull request merged**